client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,280 @@
+	discount_value: 10,
+	discount_percentage: 0,
+	discount_currency: 'USD',
+};
+
+const initialState = {
+	sites: { items: {} },
+	siteSettings: { items: {} },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {
+				1: [ testCoupon1, testCoupon2, testCoupon3 ],
+			},
+		},
+		productsList: {
+			items: {},
+		},
+	},
+};
+const closeDialog = () => {};
+
+function WrappedRecurringPaymentsCouponAddEditModal( props ) {
+	const mockStore = configureStore();
+	const store = mockStore( initialState );
+
+	return (
+		<Provider store={ store }>
+			<RecurringPaymentsCouponAddEditModal { ...props } />
+		</Provider>
+	);
+}
+
+describe( 'RecurringPaymentsCouponAddEditModal', () => {
+	let modalRoot;
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+		getconnectedAccountDefaultCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountDefaultCurrencyForSiteId
+		);
+		getconnectedAccountMinimumCurrencyForSiteId.mockImplementation(
+			mockGetConnectedAccountMinimumCurrencyForSiteId
+		);
+
+		modalRoot = document.createElement( 'div' );
+		modalRoot.setAttribute( 'id', 'wpcom' );
+		document.body.appendChild( modalRoot );
+		Modal.setAppElement( modalRoot );
+	} );
+
+	afterEach( () => {
+		unmountComponentAtNode( modalRoot );
+		document.body.removeChild( modalRoot );
+		[ ...document.getElementsByClassName( 'ReactModalPortal' ) ].map( ( el ) => {
+			document.body.removeChild( el );
+		} );
+		modalRoot = null;
+		global.Date = RealDate;
+	} );
+
+	test( 'should render empty form when no coupon is provided', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		expect( document.body ).toMatchSnapshot();
+	} );
+
+	test( 'should generate a new coupon code when the random button is clicked', () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		const randomButton = screen.getByText( 'Random' );
+		expect( screen.getByLabelText( 'Coupon code' ).value ).toEqual( '' );
+		act( () => {
+			randomButton.click();
+		} );
+		const generatedCode = screen.getByLabelText( 'Coupon code' ).value;
+		expect( generatedCode ).not.toEqual( '' );
+		expect( generatedCode.length ).toBeGreaterThan( 3 );
+	} );
+
+	test( 'should show discount amount input when discount type is amount', async () => {
+		mockDate( '2023-06-23T14:23:44z' );
+		render(
+			<WrappedRecurringPaymentsCouponAddEditModal closeDialog={ closeDialog } coupon={ {} } />
+		);
+		expect( screen.getByRole( 'option', { name: 'Amount' } ).selected ).toBe( false );
+		expect( screen.getByRole( 'option', { name: 'Percentage' } ).selected ).toBe( true );
+		expect( screen.getByLabelText( 'Amount' ) ).toMatchSnapshot();
```

Comment:
It looks like we could avoid using `toMatchSnapshot` here in favor of a more explicit (and simpler) assertion. I'd avoid `toMatchSnapshot` whenever possible.